lab_2_tokenize_by_bpe/main.py

```diff
@@ -47,6 +97,32 @@ def merge_tokens(
     :param pair: a pair of tokens to be merged
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not(
+        isinstance(word_frequencies, dict)
+        and isinstance(pair, tuple)
+    ):
+        return None
+
+    merged_dict = {}
+
+    for word_tokens in word_frequencies:
+        word_tokens_list = list(word_tokens)
+        if pair[1] in word_tokens:
+            pair_token_index = []
+            for index in range(len(word_tokens) - 1):
```

Comment:
Too dificult.. try to find out easier solution

Reply:
don't really get what exactly to change

Reply:
You have two loops - the first to find indicies of the pair in word, the second- to replace the split tokens with merged token. Try to do it in one loop